public/index.php

```diff
@@ -43,6 +43,13 @@ function getRedirection($redirect_uri)
 		case '/exercises/new':
 			createAnExercises();
 			break;
+		case (preg_match('/^\/exercises\/([0-9]+)\?exercise%5Bstatus%5D=(answering|closed)$/A', $_SERVER['REQUEST_URI'], $output_array) ? true : false):
```

Comment:
You should use variable $redirect_uri instead of $_SERVER['REQUEST_URI'] and all the "GETTER" attribute should be not checked here
It should be check in the controller with the supervariable $_GET